api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -68,23 +69,58 @@ public interface Snapshot extends Serializable {
    * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
    *
    * @return a list of ManifestFile
+   * @deprecated since 1.0.0 - Use {@link Snapshot#allManifests(FileIO)} instead.
    */
+  @Deprecated
   List<ManifestFile> allManifests();
 
+  /**
+   * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
+   * <p>
+   * The FileIO instance is passed in for cases where the correct FileIO to use is not known when the
+   * TableMetadata (and thus BaseSnapshot implementations) is not known, such as parsing JSON for the
+   * RESTCatalog or for highly sessionzed applications like Trino or shared Spark servers.
+   *
+   * @param io the {@link FileIO} instance to be used for reading Snapshot manifests, if not already cached
```

Comment:
How about `a FileIO instance for reading files in object storage`?

Reply:
I'll go with `a FileIO instance for reading files from storage` so as not to confuse any potential HDFS users. Trivial difference but let me know if you disagree